### PR TITLE
Create CreateClaimantOnEvent service class and add logic to create claimant if needed

### PR DESCRIPTION
**Pull request by buddylove954**
Resolves [Create CreateClaimantOnEvent service class and add logic to create claimant if needed](https://jira.devops.va.gov/browse/APPEALS-40953)

# Description
This pull request adds the Events::CreateClaimantOnEvent service class under the services/events/ directory, implementing logic to handle claimant creation for events. It checks if a veteran is also a claimant, creates a new person if the claimant doesn't exist, and updates cache with the latest information. It's called from the DecisionReviewCreated service class, returning the new claimant record ID and associating any new records created as BackFill records to the event. Additionally, it includes comprehensive RSPEC scenarios to validate the implemented functionality.

## Acceptance Criteria

- [x] Build out CreateClaimantOnEvent Service class that will be in services/events/ folder
- [x] This service class will have the namespace of Events::CreateClaimantOnEvent , use module to wrap class and namespace
- [x] Implement Logic to check if veteran is also a claimant.
- [x] Implement logic for creating new person if claimant does not exist. No need of calling BGS to create claimant, this should have been done in consumer app.
- [x] Make sure to update cache with latest up to date info
- [x] Make a call from the DecisionReviewCreated Service class to this service class before processing any records .
- [x] Return back the New claimant record id in CF to DecisionReviewCreated calls and store in variable for future use.
- [x] Record any new records created as BackFill records associated to Event.
- [x] Create RSPEC scenarios for all above AC.
- [x] update event record if person does not exist

## Testing Plan

1. Go to [Test Plan 42086 for APPEALS-40953](https://jira.devops.va.gov/browse/APPEALS-42086)
2. Go to [XRAY for APPEALS-40953](https://jira.devops.va.gov/secure/XrayExecuteTest!default.jspa?testExecIssueKey=APPEALS-42087&testIssueKey=APPEALS-42086)

- [ ] For feature branches merging into master: Was this deployed to UAT?

# Best practices
## Code Documentation Updates
- [x] Add or update code comments at the top of the class, module, and/or component.

### Test Coverage
Did you include any test coverage for your code? Check below:
- [x] RSpec

### Code Climate
Your code does not add any new code climate offenses? If so why?
- [x] No new code climate issues added

### CODE REVIEW

File: app/services/events/create_claimant_on_event.rb [hunk 1]
```diff
@@ -0,0 +1,21 @@
+# frozen_string_literal: true
+
+class Events::CreateClaimantOnEvent
+  class << self
+    def process(event:, claimant_attributes:)
+      claimant_attributes = OpenStruct.new(claimant_attributes)
+      if claimant_attributes.is_veteran_claimant
+        event.reference_id
```
Comment (jtsangVA) [inline app/services/events/create_claimant_on_event.rb:8]: What is this check doing?

File: app/services/events/create_claimant_on_event.rb [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+      else
+        claimant = Claimant.create!(
```
Comment (jtsangVA) [inline app/services/events/create_claimant_on_event.rb:10]: You also need to add logic to check if the Claimant already exists in Caseflow

File: app/services/events/create_claimant_on_event.rb [hunk 3]
```diff
@@ -0,0 +1,21 @@
+# frozen_string_literal: true
+
+class Events::CreateClaimantOnEvent
+  class << self
+    def process(event:, claimant_attributes:)
+      claimant_attributes = OpenStruct.new(claimant_attributes)
+      if claimant_attributes.is_veteran_claimant
```
Comment (jtsangVA) [inline app/services/events/create_claimant_on_event.rb:7]: We're not receiving anything with that name (`is_veteran_claimant`)

The boolean will be `veteran_is_not_claimant` and that's part of the claim_review

File: app/services/events/create_claimant_on_event.rb [hunk 4]
```diff
@@ -0,0 +1,21 @@
+# frozen_string_literal: true
+
+class Events::CreateClaimantOnEvent
+  class << self
+    def process(event:, claimant_attributes:)
+      claimant_attributes = OpenStruct.new(claimant_attributes)
```
Comment (nkutub) [inline app/services/events/create_claimant_on_event.rb:6]: why are converting to object?

File: app/services/events/create_claimant_on_event.rb [hunk 5]
```diff
@@ -0,0 +1,21 @@
...
+        claimant = Claimant.find_or_create_by!(
+          name_suffix: claimant_attributes.name_suffix,
+          participant_id: claimant_attributes.participant_id,
+          payee_code: claimant_attributes.payee_code,
+          type: claimant_attributes.type
+        )
+        EventRecord.create!(event: event, backfill_record: claimant)
+        claimant.id
+      else
+        event.reference_id
```
Comment (jtsangVA) [inline app/services/events/create_claimant_on_event.rb:17]: Why are you returning the event ref id?

Comment (jtsangVA) [reply app/services/events/create_claimant_on_event.rb:17]: `event.reference_id` is the primary key of the Event in the consumer app. 

`t.string "reference_id", null: false, comment: "Id of Event Record being referenced within the Appeals Consumer Application"`

File: app/services/events/create_claimant_on_event.rb [hunk 6]
```diff
@@ -0,0 +1,17 @@
+# frozen_string_literal: true
+
+class Events::CreateClaimantOnEvent
+  class << self
+    def process(event:, vbms_claimant:, decision_review:)
```
Comment (nkutub) [inline app/services/events/create_claimant_on_event.rb:5]: this needs to be process!

File: app/services/events/create_claimant_on_event.rb [hunk 7]
```diff
@@ -0,0 +1,17 @@
...
+      if vbms_claimant.claim_review.veteran_is_not_claimant
+        claimant = Claimant.find_or_create_by!(
+          decision_review: decision_review,
+          participant_id: vbms_claimant.claimant.participant_id,
+          payee_code: vbms_claimant.claimant.payee_code
+        )
+        EventRecord.create!(event: event, backfill_record: claimant)
+        claimant.id
```
Comment (nkutub) [inline app/services/events/create_claimant_on_event.rb:13]: just return claimant not id

**Review by nkutub — APPROVED** [review]

**Pull request merged**